framework/decode/vulkan_replay_dump_resources_compute_ray_tracing.cpp

```diff
@@ -1140,18 +1310,34 @@ VkResult DispatchTraceRaysDumpingContext::DumpImmutableDescriptors(uint64_t qs_i
 
     DumpedDescriptors& dumped_descriptors = is_dispatch ? dispatch_dumped_descriptors : trace_rays_dumped_descriptors;
 
+    const DispatchParameters*  disp_params;
+    const TraceRaysParameters* tr_params;
+
     if (is_dispatch)
     {
-        const auto disp_params = dispatch_params.find(cmd_index);
-        assert(disp_params != dispatch_params.end());
-        for (const auto& desc_set : disp_params->second.referenced_descriptors)
+        const auto disp_params_entry = dispatch_params.find(cmd_index);
+        GFXRECON_ASSERT(disp_params_entry != dispatch_params.end());
+
+        tr_params   = nullptr;
+        disp_params = &disp_params_entry->second;
+    }
+    else
+    {
+        const auto tr_params_entry = trace_rays_params.find(cmd_index);
+        GFXRECON_ASSERT(tr_params_entry != trace_rays_params.end());
+
+        disp_params = nullptr;
+        tr_params   = &tr_params_entry->second;
+    }
+
+    if (is_dispatch)
```

Comment:
Why is this separate from the previous `if-else` clause?

Reply:
I think some leftovers from this PR's iterations. They can be merged